Add Pulsar configuration to unsubscribe on close

aeons opened the pull request:
This PR adds an `UnsubscribeOnClose(bool)` to `PulsarTransportExtension` to set whether the Pulsar listener should unsubscribe when it's closed.

The default is `true` to keep the current behavior.

Ideally, I would prefer this (and enable requeue) to default to false and be opt-in, but that would be a breaking change.

### Motivation
We have a setup where subscriptions are created by an external system, and due to access rights limitations, our consumer can't create subscriptions but can delete them.
So we need a way to have the subscription outlive the listener.

jeremydmiller commented:
@aeons Looks good. Couple points though:

* You could add another "all Pulsar listeners should be auto-closed" off of WolverineOptions.Policies w/ an extension method
* Be nice to have a little bit in the docs
* Even if it's just a smoke test, add some kind of test first?

aeons commented:
>You could add another "all Pulsar listeners should be auto-closed" off of WolverineOptions.Policies w/ an extension method

Yeah, that makes sense. I'll see if I can figure it out.

Should I add one for `EnableRequeue` as well?

> Be nice to have a little bit in the docs

Sure.

> Even if it's just a smoke test, add some kind of test first?

I'll try :)

aeons commented:
Re: testing, I can't really figure out how to verify that the subscription is not unsubscribed at the pulsar level.

I can add some tests that the configuration extensions do apply as expected.

aeons commented:
I've added `IPolicies` extension methods, added some tests that the configuration is applied as expected and updated the docs.

I cannot figure out a good way to test the actual functionality, so if you have any ideas, please let me know.
